integration-tests/src/main/java/com/example/ModuleClassHierarchy.java

```diff
@@ -0,0 +1,23 @@
+package com.example;
+
+import dagger.Binds;
+import dagger.Component;
+import dagger.Module;
+import dagger.Provides;
+
+@Component(modules = ModuleClassHierarchy.Module1.class)
+interface ModuleClassHierarchy {
+  CharSequence string();
+
+  @Module
+  abstract class Module1 extends BaseModule {
+    @Provides static String string() {
+      return "foo";
+    }
+  }
+
+  @Module
+  abstract class BaseModule {
```

Comment:
I think the `@Module` annotation on the base classes is not necessary, because it's not used in `modules =`

Reply:
The compiler requires all types with providers to contain it